src/main/java/nu/studer/teamcity/buildscan/internal/cleanup/CustomDataStorageBuildScanCleaner.java

```diff
@@ -0,0 +1,88 @@
+package nu.studer.teamcity.buildscan.internal.cleanup;
+
+import jetbrains.buildServer.serverSide.SBuildServer;
+import jetbrains.buildServer.serverSide.SQLRunner;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+import static nu.studer.teamcity.buildscan.internal.CustomDataStorageBuildScanDataStore.BUILD_SCAN_STORAGE_ID;
+
+@SuppressWarnings("deprecation")
+public final class CustomDataStorageBuildScanCleaner {
+
+    private static final int BATCH_SIZE = 10;
+    private static final String DELETE_CUSTOM_DATA_BODY_SQL = "delete from custom_data_body where id = ?";
+    private static final String DELETE_FROM_CUSTOM_DATA_SQL = "delete from custom_data where data_id = ?";
+    private final SBuildServer server;
+
+    public CustomDataStorageBuildScanCleaner(SBuildServer buildServer) {
+        server = buildServer;
+    }
+
+    public Integer count() {
+        return server.getSQLRunner().runSql(connection -> {
+            try (Statement statement = connection.createStatement()) {
+                ResultSet resultSet = statement.executeQuery("select count(data_id) from custom_data where data_key='" + BUILD_SCAN_STORAGE_ID + "'");
+                resultSet.next();
+                return resultSet.getInt("count");
```

Comment:
This did not work for me locally unless I defined an alias for the count (which I did, as you can see on the merged branch).